src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -594,11 +594,16 @@ private IEnumerator RecordFrame()
                     }
                     catch (Exception e)
                     {
-                        RGDebug.LogException(e, "Exception capturing state for frame");
+                        RGDebug.LogException(e, $"Exception capturing state for tick # {_tickNumber}");
                     }
 
+                    var didQueue = 0;
+
                     if (jsonData != null)
                     {
+                        // save this off because we're about to operate on a different thread :)
+                        var currentTickNumber = _tickNumber;
```

Comment:
This guy right here... this is the actual fix for the bug.  Need a copy of the `_tickNumber` for the async code to use.  It was currently getting the `_tickNumber` field which had already been updated by the next frame processing.

Reply:
Good catch!